frontend/src/booking/BookingPage.js

```diff
@@ -44,6 +44,7 @@ const BookingPage = (props) => {
                 <Route path="/" component={() => <ContentContainer type="time" mainHistory={history} />} />
               </Switch>
             </Router>
+            <Prompt when={true} message="Are you sure you want to leave before finishing?"/>
```

Comment:
I think this text could be a little more explanatory, something like "Are you sure you want to leave before completing your booking? No changes will be saved.". Otherwise, it might be unclear to a user what they have or have not finished.

Reply:
I have added more explanatory text as suggested. The message comes from the base url of the page so in a production environment it would be more suitable (i.e. a popup from github would say 'github.com says').